pkg/csi/service/common/authmanager.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/common/cns-lib/vsphere"
+	cnsvsphere "sigs.k8s.io/vsphere-csi-driver/pkg/common/cns-lib/vsphere"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+// AuthorizationService exposes interfaces to support authorization check on datastores and get datastores which need to be
+// ignored by create volume
+type AuthorizationService interface {
+
+	// GetDatastoreIgnoreMapForBlockVolumes returns a map which maps datastore url to datastore info which user does not have Datastore.FileManagement privilege
+	// returns an error if not able to retrieve the map
+	GetDatastoreIgnoreMapForBlockVolumes(ctx context.Context) (map[string]*cnsvsphere.DatastoreInfo, error)
+}
+
+// AuthManager maintains an internal map to track the datastores that need to be ignored by create volume
+type AuthManager struct {
+	// map the datastore url to datastore info which need to be ignored when invoking CNS CreateVolume API
+	datastoreIgnoreMapForBlockVolumes map[string]*cnsvsphere.DatastoreInfo
```

Comment:
This will be renamed from `datastoreIgnoreMapForBlockVolumes` to `datastoreMapForBlockVolumes`

Reply:
Same as above. I will not make this change.